plugins/woocommerce/src/Blocks/BlockTypesController.php

```diff
@@ -164,7 +197,9 @@ public function add_data_attributes( $content, $block ) {
 		 */
 		$allowed_blocks = (array) apply_filters( '__experimental_woocommerce_blocks_add_data_attributes_to_block', array() );
 
-		if ( ! in_array( $block_namespace, $allowed_namespaces, true ) && ! in_array( $block_name, $allowed_blocks, true ) ) {
+		$block_has_woo_parent = in_array( $block_name, array_keys( $this->registered_blocks_with_woocommerce_parents ), true );
+
+		if ( ! $block_has_woo_parent && ! in_array( $block_namespace, $allowed_namespaces, true ) && ! in_array( $block_name, $allowed_blocks, true ) ) {
```

Comment:
I would move this complicated conditional into a separate method for clarity.

Reply:
Done, I moved a lot of this into `block_should_have_data_attributes` seems cleaner now.